### PR TITLE
Return 404 when requested resource not found/inaccessible

### PR DESCRIPTION
**Pull request by acrellin**

### CODE REVIEW

File: app/handlers/base.py [hunk 1]
```diff
@@ -309,6 +309,8 @@ def error(self, message, data={}, status=400, extra={}):
         if not (status == 404 and self.request.method == "HEAD"):
             log(f'Error response returned by [{self.request.path}]: [{message}]')
 
+        if message.startswith("Access Error: Invalid"):
```
Comment (stefanv) [inline app/handlers/base.py:312]: Should this live in `write_error` instead?

Comment (acrellin) [reply app/handlers/base.py:312]: I don't think it matters -- `write_error` just calls `error`. If we were just checking for the `AccessError` exception type it'd make more sense to have it there, but we don't want to return 404 for all `AccessError`s raised -- just those with this error message (there are other cases where it's raised where 404 doesn't make as much sense).

Comment (stefanv) [reply app/handlers/base.py:312]: That's a very specific thing to catch and handle in a special way.  Is this generic enough?  Is there a way to make it more generic (such as raising an ObjectNotFound exception and handling that differently?).

**Comment by acrellin** [comment]
Sure, that sounds good. Will update.

**Comment by acrellin** [comment]
@stefanv updated, ready for re-review

**Comment by stefanv** [comment]
I like it!

**Pull request merged**